Automattic/simplenote-electron
# UI Spring Cleaning: Note info panels

### Fix

* Split note info component into two components, NoteInfo and NoteActions.
* Remove publish note info from Share dialog
* Remove redundant icons from the note toolbar

Fixes #2569 

Known issues:
- There's a bug where if you check a checkbox repeatedly. This has to do with updates to the note coming back from GAE. The timing can be off and GAE will send an update basically undoing one of the changes in-app. This isn't something we are going to fix right now as it would involve substantial work in GAE. Thanks, @dmsnell for helping track this down.
- Dark mode colors aren't 100% and will be updated separately. 

**Screenshots**

**Before:**
<details>
<summary>Light Mode</summary>
<img width="839" alt="Screen Shot 2021-03-22 at 9 56 02 AM" src="https://user-images.githubusercontent.com/1326294/111993185-14789700-8af5-11eb-9115-90d5557626e9.png">
</details>
<details>
<summary>Dark Mode</summary>
<img width="835" alt="Screen Shot 2021-03-22 at 9 56 15 AM" src="https://user-images.githubusercontent.com/1326294/111993193-16daf100-8af5-11eb-8082-1749c5e77db0.png">
</details>

**After:**

**Light Mode**
<details>
<summary>Note info w/ references</summary>
<img width="868" alt="Screen Shot 2021-03-22 at 9 50 56 AM" src="https://user-images.githubusercontent.com/1326294/111993517-79cc8800-8af5-11eb-93a5-d4af6875a2e7.png">
</details>
<details>
<summary>Note info no references</summary>
<img width="867" alt="Screen Shot 2021-03-22 at 9 51 09 AM" src="https://user-images.githubusercontent.com/1326294/111993539-7f29d280-8af5-11eb-9500-425f60cea870.png">
</details>
<details>
<summary>Note Actions</summary>
<img width="864" alt="Screen Shot 2021-03-22 at 9 52 21 AM" src="https://user-images.githubusercontent.com/1326294/111993618-98328380-8af5-11eb-80d8-21c6cd3ee403.png">
</details>

**Dark Mode**
<details>
<summary>Note info w/ references</summary>
<img width="862" alt="Screen Shot 2021-03-22 at 9 50 39 AM" src="https://user-images.githubusercontent.com/1326294/111993962-fc554780-8af5-11eb-9f39-6f6fd15bef6e.png">
</details>
<details>
<summary>Note info no references</summary>
<img width="867" alt="Screen Shot 2021-03-22 at 9 51 23 AM" src="https://user-images.githubusercontent.com/1326294/111993810-d039c680-8af5-11eb-866e-6ee30395ed77.png">
</details>
<details>
<summary>Note Actions</summary>
<img width="866" alt="Screen Shot 2021-03-22 at 9 51 56 AM" src="https://user-images.githubusercontent.com/1326294/111993912-ea73a480-8af5-11eb-8701-19315f5a376c.png">
</details>

### Test

1. Open a note
2. Notice new note toolbar 
3. Click on the new info `(i)` icon
4. Notice the new info dialog 
5. Click the new actions `(...)` icon
6. Ensure all options work as expected
7. Smoke test in light and dark mode ensuring styles are correct

### Release Notes

- Updates the note toolbar and note info sidebar to be two dialogs, one for note info, and one for note actions.
